schism/compiler.ss

```diff
@@ -529,11 +528,11 @@
   (define (parse-libraries libs)
     (let ((imported-envs (map make-export-environment libs)))
       (let ((first (parse-library (car libs) imported-envs))
-            (imported-functions (fold-left (lambda (functions lib)
-                                             (append functions
-                                                     (cdr (parse-library lib imported-envs))))
-                                           '()
-                                           (cdr libs))))
+            (imported-functions (fold-right (lambda (lib functions)
+                                              (append (cdr (parse-library lib imported-envs))
+                                                      functions))
+                                            '()
+                                            (cdr libs))))
```

Comment:
I think this was quadratic because the `functions` list gets traversed repeatedly in `append`